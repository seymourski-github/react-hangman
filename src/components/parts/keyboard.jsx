import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Label } from '../common';

export const Keyboard = props => {
  const { action, keys, id, handleAction, label } = props;

  const buttons = keys.map(key =>
    <Button {...key}
      key={key.name}
      action={action}
      handleAction={handleAction}
    />);

  return (
    <Container id={id}>
      <Label text={label} flexItem="100%" />
      {buttons}
    </Container>
  );
};

Keyboard.propTypes = {
  action: PropTypes.string,
  id: PropTypes.string.isRequired,
  keys: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  })).isRequired,
  label: PropTypes.string,
  handleAction: PropTypes.func
};

Keyboard.defaultProps = {
  action: 'ACTION_SENT_WITH_HANDLE_ACTION',
  label: 'Choose a letter',
  handleAction: () => console.log('handleAction button fail')
};
