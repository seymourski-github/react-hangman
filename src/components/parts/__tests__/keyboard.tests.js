import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Keyboard } from '../keyboard';

const props = {
  id: 'my-keyboard-panel',
  keys: [{
    name: 'a'
  }, {
    name: 'b'
  }, {
    name: 'c'
  }]
};

describe('<Keyboard /> Component', function() {

  const component = renderer.create(<Keyboard {...props} />);
  let wrapper = shallow(<Keyboard {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
