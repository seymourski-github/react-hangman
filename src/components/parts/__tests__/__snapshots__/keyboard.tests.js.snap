// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keyboard /> Component renders and matches our snapshot 1`] = `
<div
  className="container row"
  id="my-keyboard-panel"
  style={undefined}
>
  <label
    className="label"
    disabled={false}
    hidden={false}
    htmlFor={undefined}
    style={
      Object {
        "alignSelf": "",
        "flex": "100%",
        "visibility": "visible",
      }
    }
  >
    Choose a letter
  </label>
  <button
    className="button"
    disabled={false}
    name="a"
    onClick={[Function]}
    style={null}
    type="button"
    value=""
  >
    a
  </button>
  <button
    className="button"
    disabled={false}
    name="b"
    onClick={[Function]}
    style={null}
    type="button"
    value=""
  >
    b
  </button>
  <button
    className="button"
    disabled={false}
    name="c"
    onClick={[Function]}
    style={null}
    type="button"
    value=""
  >
    c
  </button>
</div>
`;
